core/src/banking_stage/vote_storage.rs

```diff
@@ -209,32 +148,40 @@ impl VoteStorage {
             panic!("Gossip vote thread should not be processing transactions");
         }
 
-        let should_process_packet =
-            |packet: &Arc<ImmutableDeserializedPacket>, payload: &mut ConsumeScannerPayload| {
-                consume_scan_should_process_packet(&bank, banking_stage_stats, packet, payload)
-            };
-
         // Based on the stake distribution present in the supplied bank, drain the unprocessed votes
         // from each validator using a weighted random ordering. Votes from validators with
         // 0 stake are ignored.
         let all_vote_packets = self
             .latest_unprocessed_votes
             .drain_unprocessed(bank.clone());
 
-        let mut scanner = create_consume_multi_iterator(
-            &all_vote_packets,
-            slot_metrics_tracker,
-            should_process_packet,
-        );
-
         let deprecate_legacy_vote_ixs = self
             .latest_unprocessed_votes
             .should_deprecate_legacy_vote_ixs();
 
-        while let Some((packets, payload)) = scanner.iterate() {
-            let vote_packets = packets.iter().map(|p| (*p).clone()).collect_vec();
-
-            if let Some(retryable_vote_indices) = processing_function(&vote_packets, payload) {
+        let mut payload = ConsumeScannerPayload {
+            reached_end_of_slot: false,
+            sanitized_transactions: Vec::with_capacity(UNPROCESSED_BUFFER_STEP_SIZE),
+            slot_metrics_tracker,
+            error_counters: TransactionErrorMetrics::default(),
+        };
+
+        for chunk in all_vote_packets.chunks(UNPROCESSED_BUFFER_STEP_SIZE) {
+            let mut vote_packets: Vec<Arc<ImmutableDeserializedPacket>> =
```

Comment:
let's define this outside the loop, and clear it. no need to re-allocate each iteration.
since we know capacity, could also just use an ArrayVec here to avoid allocating entirely

Reply:
fine to leave that for follow-up, since the payload will be removed as well

Reply:
well i meant just the arrayvec part of it; defining the vec inside the loop shouldn't be left in